docs(template): add the vanilla template (#1742)
latticexyz/mud: next-docs/pages/templates/typescript/vanilla.mdx

# Vanilla

> The vanilla template uses standard HTML and JavaScript to communicate with the user.

Reviewer: ```suggestion
The vanilla template uses plain HTML and JavaScript for the web client.
```

## Onchain code

This template uses [the standard `Counter` onchain code](./contracts).

## Offchain code

The files you are likely to need to change in the offchain code are:

- [`packages/client/index.html`](https://github.com/latticexyz/mud/blob/main/templates/vanilla/packages/client/index.html), which controls the information displayed to the user.
- [`packages/client/src/index.ts`](https://github.com/latticexyz/mud/blob/main/templates/vanilla/packages/client/src/index.ts), which contains the setup code.
  It is the place where you register handlers that are called when the value in a component (basically, a table) changes.
- [`packages/client/src/mud/createSystemCalls.ts`](https://github.com/latticexyz/mud/blob/main/templates/vanilla/packages/client/src/mud/createSystemCalls.ts), which is where you write code that performs calls to an onchain system.

### index.html

This is the HTML displayed to the user.
The lines that are relevant to MUD are:

```html
<script type="module" src="/src/index.ts"></script>
```

Import the script that runs most of the application.

```html
<div>Counter: <span id="counter">0</span></div>
```

> Create [an HTML span](https://www.w3schools.com/tags/tag_span.asp) that JavaScript code can update.

Reviewer: ```suggestion
Create [an HTML span](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/span) that JavaScript code can update.
```

This span will be updated every time the value of the counter changes.

```html
<button onclick="window.increment()">Increment</button>
```

A button that calls `window.increment()`, which calls the onchain system to increment the counter.

### index.ts

This is the main TypeScript file.
It calls the general setup code for MUD, and then does application-specific setup actions.

```ts
// Components expose a stream that triggers when the component is updated.
components.Counter.update$.subscribe((update) => {
  const [nextValue, prevValue] = update.value;
  console.log("Counter updated", update, { nextValue, prevValue });
  document.getElementById("counter")!.innerHTML = String(nextValue?.value ?? "unset");
});
```

Specify a function to be called when the value of `Counter` changes.

> This function emits [a log message](https://www.w3schools.com/jsref/met_console_log.asp) and then updates the `counter` span.

Reviewer: would you mind if we use the MDN docs for links like this everywhere? I personally just prefer that site over `w3schools`
```suggestion
This function emits [a log message](https://developer.mozilla.org/en-US/docs/Web/API/console/log) and then updates the `counter` span.
```
